content/topics/java-specific/oop-basics/_index.md

```diff
@@ -1,15 +1,179 @@
+
+The following Java code shows how Abstraction can be implemented:
+
+```
+//abstract parent class
+abstract class Animal{
+    //abstract method
+    public abstract void sound();
+    //regular method
+    public void sleep() {
+    System.out.println("Zzz");
+  }
+}
+//Dog class extends Animal class
+class Dog extends Animal{
+
+    public void sound(){
+        System.out.println("Woof");
+    }
+    public static void main(String args[]){
+        Animal obj = new Dog();
+        obj.sound();
+    }
+}
+ 
+``` 
+
+
+### Encapsulation
+Encapsulation is known as data-hiding.
+It is an OOP principle which states that all the attributes and behaviours (methods and functions) of an object should be grouped together in one datatype or class.
+This means that objects may be able to communicate with one another but are restricted to access some of the object's components directly.
+Publicly accessible methods are generally provided in the class so-called [accessors and mutators.](https://www.cs.colostate.edu/~cs161/Fall12/labs/lab2/bookgetset.html)) 
+
+The following Java code shows how Encapsulation can be implemented:
+
+```
+public class Employee{
+
+    //private data member  
+    private String name;
+
+    //getter method for name  
+    public String getName(){
+        return name;
+    }Encapsulation is known as data-hiding.
+It is an OOP principle which states that all the attributes and behaviours (methods and functions) of an object should be grouped together in one datatype or class.
+This means that objects may be able to communicate with one another but are restricted to access some of the object's components directly.
+Publicly accessible methods are generally provided in the class (so-called [accessors and mutators.]
+    }
+
+    public static void main(String args[]) { 
+        //creating instance of the encapsulated class
+        Employee e=new Employee();
+        //setting value in the name member  
+        e.setName("Mbali");
+        //getting value of the name member  
+        System.out.println(e.getName());
+
+    }
+}  
+```
+[Abstraction vs Encapsulation](https://1.bp.blogspot.com/-ECYNAUTGGMk/WPQeY4EpFtI/AAAAAAAAIX8/j-Ji8N_mDz8-d72SasgNPnQD-nIlw-kiACLcB/s1600/Abstraction%2Bvs%2BEncapsulation%2B2.jpg)
+
+### Inheritance
+This OOP principle allows one class to inherit all the attributes and behaviors of another class.
+Inheritance has a hierarchy  relationship between superclasses (base classes )and subclasses
+
+The following Java code shows how Inheritance can be implemented:
+```
+//superclass
+public class Vehicle {
+    void printType(){
+        System.out.println("I am a Vehicle");
+    }
+}
+
+public class Car extends  Vehicle {
+    //Override method 
+    @Override
+    void printType() {
+        //call method in super class
+        super.printType();
+        System.out.println("I am a car");
+    }
+
+
+    public static void main(String[] args) {
```

Comment:
are you aware that your main function is inside your car class o.O 
This is not good code, it will teach bad haabits. 

I think you and I should talk about OOP